workflow/rules/purecn.smk

```diff
@@ -0,0 +1,112 @@
+    container:
+        config.get("purecn_coverage", {}).get("container", config["default_container"])
+    conda:
+        "../envs/purecn_coverage.yaml"
+    message:
+        "{rule}: Calculate coverage for {wildcards.sample}"
+    shell:
+        "(Rscript $PURECN/Coverage.R "
+        "--out-dir=cnv_sv/purecn_coverage "
+        "--bam={input.bam} "
+        "--intervals={params.intervals} "
+        "--cores={threads} "
+        "{params.extra}) &> {log}"
+
+
+rule purecn:
+    input:
+        unpack(get_purecn_inputs),
+        vcf="snv_indels/gatk_mutect2/{sample}_{type}.merged.softfiltered.vcf.gz",
+        tbi="snv_indels/gatk_mutect2/{sample}_{type}.merged.softfiltered.vcf.gz.tbi",
+    output:
+        temp(
+            expand(
+                "cnv_sv/purecn/{{sample}}_{{type}}{ext}",
+                ext=[
+                    ".csv",
+                    ".rds",
+                    ".pdf",
+                    "_dnacopy.seg",
+                    "_chromosomes.pdf",
+                    "_segmentation.pdf",
+                    "_local_optima.pdf",
+                    "_variants.csv",
+                    "_loh.csv",
+                ],
+            )
+        ),
+    params:
+        fun_segmentation=config.get("purecn", {}).get("fun_segmentation", ""),
+        genome=config.get("purecn", {}).get("genome", ""),
+        interval_padding=config.get("purecn", {}).get("interval_padding", ""),
+        extra=get_purecn_extra,
+    log:
+        "cnv_sv/purecn/{sample}_{type}.output.log",
+    benchmark:
+        repeat(
+            "cnv_sv/purecn/{sample}_{type}.output.benchmark.tsv",
+            config.get("purecn", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("purecn", {}).get("threads", config["default_resources"]["threads"])
+    resources:
+        mem_mb=config.get("purecn", {}).get("mem_mb", config["default_resources"]["mem_mb"]),
+        mem_per_cpu=config.get("purecn", {}).get("mem_per_cpu", config["default_resources"]["mem_per_cpu"]),
+        partition=config.get("purecn", {}).get("partition", config["default_resources"]["partition"]),
+        threads=config.get("purecn", {}).get("threads", config["default_resources"]["threads"]),
+        time=config.get("purecn", {}).get("time", config["default_resources"]["time"]),
+    container:
+        config.get("purecn", {}).get("container", config["default_container"])
+    conda:
+        "../envs/purecn.yaml"
+    message:
+        "{rule}: Quantify purity/ploidy for {wildcards.sample}_{wildcards.type}"
+    shell:
+        "(Rscript $PURECN/PureCN.R "
+        "--tumor={input.tumor} "
+        "--vcf={input.vcf} "
+        "--genome={params.genome} "
+        "--fun-segmentation={params.fun_segmentation} "
+        "--interval-padding={params.interval_padding} "
+        "--sampleid={wildcards.sample} "
+        "--out=cnv_sv/purecn/{wildcards.sample}_{wildcards.type} "
+        "--force --seed=1337 "
```

Comment:
Is it possible that one would like to be able to change "--seed"? If that's the case we should maybe add a param for it with a default value.

Reply:
If it's no point of allowing a user to modify it you can leave it as it is.

Reply:
OK, will leave it for now